src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -2786,11 +2786,11 @@ private void WriteToServerInternalRestContinuedAsync(BulkCopySimpleResultSet int
 
                     try
                     {
-                        CleanUpStateObjectOnError();
+                        CleanUpStateObject(isCancelRequested: false);
                     }
                     catch (Exception cleanupEx)
                     {
-                        Debug.Fail("Unexpected exception during CleanUpstateObjectOnError (ignored)", cleanupEx.ToString());
+                        Debug.Fail("Unexpected exception during CleanUpstateObject (ignored)", cleanupEx.ToString());
```

Comment:
Might as well fix the casing while you are changing this:
CleanUpstateObject -> CleanUp**S**tateObject